guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
+| ----------- | ------------------------- | ------------------------------- |
+| `:path`     | Complete path to the file | `"/var/app/exports/report.csv"` |
+
+In this case, the payload will always contain the `:path` key, but will also
+contain the `:filename`, and `:disposition` keys which are passed to the
+`send_file` method.
 
 #### `send_data.action_controller`
 
-`ActionController` does not add any specific information to the payload. All options are passed through to the payload.
+The event is emitted when a controller sends raw data with `send_data`.
+
+For example:
+
+```ruby
+send_data "Hello, world!",
+  filename: "report.txt",
+  type: "text/plain",
+  disposition: "attachment"
+```
+
+`ActionController` does not add any specific information to the payload. All
+options that you pass into `send_data` are passed through to the payload. Hence,
+in the example above, the payload will contain the keys `:filename`, `:type`,
+and `:disposition`.
 
 #### `redirect_to.action_controller`
 
-| Key         | Value                                    |
-| ----------- | ---------------------------------------- |
-| `:status`   | HTTP response code                       |
-| `:location` | URL to redirect to                       |
-| `:request`  | The [`ActionDispatch::Request`][] object |
+The event is emitted when a redirect response is issued by a controller using
+`redirect_to`.
+
+For example:
 
 ```ruby
-{
-  status: 302,
-  location: "http://localhost:3000/posts/new",
-  request: <ActionDispatch::Request:0x00007ff1cb9bd7b8>
-}
+class PostsController < ApplicationController
+  def old
+    redirect_to posts_url
+  end
+end
 ```
 
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key | Description                       | Example                             |
+| ----------- | --------------------------------- | ----------------------------------- |
+| `:status`   | HTTP response code                | `302`                               |
+| `:location` | URL to redirect to                | `"http://localhost:3000/posts/new"` |
+| `:request`  | The [`ActionDispatch::Request`][] | `#<ActionDispatch::Request ...>`    |
+
 #### `halted_callback.action_controller`
 
-| Key       | Value                         |
-| --------- | ----------------------------- |
-| `:filter` | Filter that halted the action |
+The event is emitted when a before/around/after callback halts the action chain.
+
+For example:
 
 ```ruby
-{
-  filter: ":halting_filter"
-}
+class PostsController < ApplicationController
+  before_action :require_login
+end
 ```
 
+The event payload (`event.payload`) includes the following keys (with typical
+example values).
+
+| Payload Key | Description                   | Example            |
+| ----------- | ----------------------------- | ------------------ |
+| `:filter`   | Filter that halted the action | `":require_login"` |
```

Comment:
Nitpick, but the usage of `filter` seems to be a remnant of controller callbacks using  `before_filter` instead of `before_action`. The payload key should probably be renamed as well in Rails.
```suggestion
| `:filter`   | Callback that halted the action | `":require_login"` |
```